crates/bevy_math/src/rotation2d.rs

```diff
@@ -0,0 +1,580 @@
+        self.sin.is_nan() || self.cos.is_nan()
+    }
+
+    /// Returns whether `self` has a length of `1.0` or not.
+    ///
+    /// Uses a precision threshold of `1e-6`.
+    #[inline]
+    pub fn is_normalized(self) -> bool {
+        let length = self.sin.hypot(self.cos);
+        length - 1.0 <= 1e-6
+    }
+
+    /// Returns `true` if the rotation is near [`Rotation2d::IDENTITY`].
+    #[inline]
+    pub fn is_near_identity(self) -> bool {
+        // Same as `Quat::is_near_identity`, but using sine and cosine
+        let threshold_angle_sin = 0.000_049_692_047; // let threshold_angle = 0.002_847_144_6;
+        self.cos > 0.0 && self.sin.abs() < threshold_angle_sin
+    }
+
+    /// Returns the angle in radians needed to make `self` and `other` coincide.
+    #[inline]
+    pub fn angle_between(self, other: Self) -> f32 {
+        (other * self.inverse()).as_radians()
+    }
+
+    /// Returns the inverse of the rotation. This is also the conjugate
+    /// of the unit complex number representing the rotation.
+    #[inline]
+    #[must_use]
+    #[doc(alias = "conjugate")]
+    pub fn inverse(self) -> Self {
+        Self {
+            cos: self.cos,
+            sin: -self.sin,
+        }
+    }
+
+    /// Performs a linear interpolation between `self` and `rhs` based on
+    /// the value `s`.
+    ///
+    /// This corresponds to computing an angle for a point at position `s` on a line drawn
+    /// between the endpoints of the arc formed by `self` and `rhs` on a unit circle.
+    ///
+    /// When `s == 0.0`, the result will be equal to `self`.  When `s == 1.0`, the result will be equal to `rhs`.
+    ///
+    /// Note that if the angles are opposite like 0 and π, the line will pass through the origin,
+    /// and the resulting angle will always be either `self` or `rhs` depending on `s`.
+    /// If `s` happens to be `0.5` in this case, a valid rotation cannot be computed, and `self`
+    /// will be returned as a fallback.
+    ///
+    /// If you would like the angular velocity to be constant, consider using [`slerp`](Self::slerp) instead.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_math::Rotation2d;
+    /// #
+    /// let rot1 = Rotation2d::IDENTITY;
+    /// let rot2 = Rotation2d::degrees(135.0);
+    ///
+    /// let result1 = rot1.lerp(rot2, 1.0 / 3.0);
+    /// assert_eq!(result1.as_degrees(), 28.675055);
+    ///
+    /// let result2 = rot1.lerp(rot2, 0.5);
+    /// assert_eq!(result2.as_degrees(), 67.5);
+    /// ```
+    #[inline]
+    pub fn lerp(self, end: Self, s: f32) -> Self {
+        Self {
+            sin: self.sin.lerp(end.sin, s),
+            cos: self.cos.lerp(end.cos, s),
+        }
+        .try_normalize()
+        // Fall back to the start rotation.
+        // This can happen when `self` and `end` are opposite angles and `s == 0.5`,
+        // because the resulting rotation would be zero, which cannot be normalized.
+        .unwrap_or(self)
+    }
```

Comment:
Can we rename this nlerp, for normalized lerp? Docs should emphasize this is more efficient than slerp, and has similar output when the difference between rotations is small.

I think making the first point a tie-breaker seems acceptable. I don't expect this will be used much when the two rotations are more than pi/2 apart, let alone approaching pi.

Reply:
I restructured the docs a bit and added the things you mentioned.

I also renamed `lerp` to `nlerp` for now. You could argue that it's a bit confusing since `slerp` should also keep the rotation normalized, assuming the input is normalized. Unity also just calls it [`Lerp`](https://docs.unity3d.com/ScriptReference/Quaternion.Lerp.html), at least for quaternions. Nalgebra seems to have both [`lerp`](https://docs.rs/nalgebra/latest/nalgebra/geometry/type.UnitQuaternion.html#method.lerp) *and* [`nlerp`](https://docs.rs/nalgebra/latest/nalgebra/geometry/type.UnitQuaternion.html#method.nlerp) though. I'm fine with either name.

Reply:
Unity also uses [`nlerp`](https://docs.unity3d.com/Packages/com.unity.mathematics@1.2/api/Unity.Mathematics.math.nlerp.html) in some places.